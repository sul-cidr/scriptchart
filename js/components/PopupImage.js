import React from "react";
import ReactDOM from "react-dom";

/* PopupImage - This class encapsulates the popup widget that appears when
 * a binarized or untrimmed letter is clicked or hovered (depending on the
 * setting from the options form).
 */

import Popup from "reactjs-popup";

class PopupImage extends React.Component {
  render() {
    return (
      <div className={"letter-image"}>
        <Popup
          trigger={this.props.triggerImage}
          position="top center"
          contentStyle={{
            width: `${this.props.width}px`, // note: images are sometimes smaller than this, but never bigger
            zIndex: 31 // Needs to occlude header, which is 30
          }}
          on={this.props.contextMode}
          arrow={false}
          mouseEnterDelay={300}
          keepTooltipInside={true}
          {...(this.props.contextMode === "click"
            // This is a hack to work around a bug in the react-popup third-party
            //  library; the library sets "position: relative" on the trigger
            //  element (the image, in our case) and doesn't remove it on close.
            // This causes an trigger image that has previously been, er, triggered
            //  to remain clickable 'through' the overlay which closes an already
            //  open popup.
            ? {
                onClose: () =>
                  ReactDOM.findDOMNode(this)
                    .querySelector("img")
                    .style.removeProperty("position")
              }
            : {})}
        >
          {this.props.contextImage}
        </Popup>
      </div>
    );
  }
}

export default PopupImage;
