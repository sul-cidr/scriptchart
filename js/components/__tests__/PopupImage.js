/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import PopupImage from "../PopupImage";

describe("PopupImage test", () => {
  it("PopupImage should match snapshot", () => {
    const component = renderer.create(<PopupImage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
