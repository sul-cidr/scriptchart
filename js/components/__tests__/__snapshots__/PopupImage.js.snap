// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupImage test PopupImage should match snapshot 1`] = `
<div
  className="letter-image"
/>
`;
